client/recordUtil.js

```diff
@@ -132,151 +28,62 @@ const normalizeBookmark = (record) => {
 }
 
 /**
- * Given a SyncRecord and a browser's matching existing object, resolve
- * objectData to the final object that should be applied by the browser.
- * @param {Object} record SyncRecord JS object
- * @param {Object=} existingObject Browser object as syncRecord JS object
- * @returns {Object|null} Resolved syncRecord to apply to browser data
+ * Keeps only most the right pairs in array for each unique objectId from recordsAndExistingObjects
  */
-const resolveRecordWithObject = (record, existingObject) => {
-  const type = record.objectData
-  if (type === 'siteSetting') {
-    return resolveSiteSettingsRecordWithObject(record, existingObject)
-  }
-  if (record.action === proto.actions.UPDATE) {
-    if (deepEqual(record[type], existingObject[type])) {
-      // no-op
-      return null
+module.exports.keepMostRight = (recordsAndExistingObjects) => {
+  var alreadySeenObjectId = new Set()
+  for (var i = recordsAndExistingObjects.length - 1; i >= 0; --i) {
+    if (recordsAndExistingObjects[i][0]) {
+      const stringObjectId = `${recordsAndExistingObjects[i][0].objectId}`
+      if (alreadySeenObjectId.has(stringObjectId)) {
+        recordsAndExistingObjects.splice(i, 1)
+      } else {
+        alreadySeenObjectId.add(stringObjectId)
+      }
     }
-    return record
-  } else if (record.action === proto.actions.DELETE) {
-    return record
-  } else {
-    throw new Error('Invalid record action')
   }
+  return recordsAndExistingObjects
 }
 
 /**
- * Given a siteSettings SyncRecord and a browser's matching existing object, resolve
- * objectData to only have the applicable fields.
- * TODO: Maybe make behavior for siteSettings same as for other types.
- * @param {Object} record SyncRecord JS object
- * @param {Object=} existingObject Browser object as syncRecord JS object
- * @returns {Object|null} Resolved syncRecord to apply to browser data
+ * Gets the record from if its timestamp greater, or action is DELETE
+ * If any of the records does not have the timestamp, choose server record
  */
-const resolveSiteSettingsRecordWithObject = (record, existingObject) => {
-  const commonFields = ['hostPattern']
-  const type = record.objectData
-  const recordFields = new Set(Object.keys(record[type]))
-  const existingFields = new Set(Object.keys(existingObject[type]))
+module.exports.getThisPairWinner = (recordAndLocalObject) => {
+  const record = recordAndLocalObject[0]
+  const object = recordAndLocalObject[1]
```

Comment:
can we rename this to `localRecord`?

Reply:
Yes, renamed at https://github.com/brave/sync/pull/353/files#diff-9df70306526678129287e667641e8ccbR54